regions/shapes/annulus.py

```diff
@@ -0,0 +1,89 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+import math
+import operator
+
+import numpy as np
+
+from astropy import wcs
+from astropy import coordinates
+from astropy import units as u
+
+from ..core import CompoundPixelRegion, CompoundSkyRegion
+from ..shapes import CirclePixelRegion, CircleSkyRegion
+
+
+class AnnulusPixelRegion(CompoundPixelRegion):
+    """
+    An annulus in pixel coordinates.
+
+    Parameters
+    ----------
+    center : :class:`~regions.core.pixcoord.PixCoord`
+        The position of the center of the annulus.
+    inner_radius : float
+        The inner radius of the annulus
+    outer_radius : float
+        The outer radius of the annulus
+    """
+
+    def __init__(self, center, inner_radius, outer_radius, meta=None, visual=None):
```

Comment:
I think there's just some DS9 regions (like text or n-annulus) that are special and don't fit in with regions like circle and polygon that can be used for analysis.
Maybe we have to find a sane way to categorize / handle that?

It's true that contains doesn't make much sense for n-annulus. DS9 has special high-performance code to generate integer-labeled masks for regions in single pixel-pass in C (see e.g. https://github.com/ericmandel/regions) and e.g. the X-ray people use it to do radial profiles in the DS9 GUI.

IMO we shouldn't let DS9 influence the design of this `regions` package too much.
Not sure what this means concretely for this class and others.
Maybe you can discuss / figure it out in Leiden?